src/DynamoCore/Search/SearchElements/ZeroTouchSearchElement.cs

```diff
@@ -22,8 +24,16 @@ public ZeroTouchSearchElement(FunctionDescriptor functionDescriptor)
             Name = displayName;
             FullCategoryName = functionDescriptor.Category;
             Description = functionDescriptor.Description;
+            Assembly = functionDescriptor.Assembly;
+
+            if (functionDescriptor.InputParameters != null)
+                inputParameters = new List<Tuple<string, string>>(functionDescriptor.InputParameters);
+            outputParameters = new List<string>() { functionDescriptor.ReturnType };
```

Comment:
Adding on to the design best practice I mentioned before, an API that returns a collection/array should ideally not return `null` to the caller. In view of that, I'd recommend you try to do this (I'm guessing `AddRange` is available on `List`?):

```csharp
inputParameters = new List<Tuple<string, string>>();
if (functionDescriptor.InputParameters != null)
    inputParameters.AddRange(functionDescriptor.InputParameters);
```